core/src/assets/login.js

```diff
@@ -32,21 +32,39 @@ function toggleOIDCForm(isVisible) {
     : 'none';
 }
 
+function toggleConfigForm(isVisible) {
+  document.querySelector('#textarea-form').style.display = isVisible
+    ? 'block'
+    : 'none';
+  document.querySelector('#file-input').style.display = isVisible
+    ? 'flex'
+    : 'none';
+}
+
+function toggleBackButton(isVisible) {
+  document.querySelector('#back-button').style.display = isVisible
+    ? 'block'
+    : 'none';
+}
+
 function toggleError(isVisible) {
   document.querySelector('#error').style.display = isVisible ? 'block' : 'none';
 }
 
-async function onKubeconfigUploaded(e) {
+function displayInitialView() {
   toggleError(false);
   toggleOIDCForm(false);
+  toggleConfigForm(true);
+  toggleBackButton(false);
+}
+function handleKubeconfig(kubeconfig, type) {
```

Comment:
You cannot handle a file 😄  
What about `handleKubeconfigAdded` or `applyKubeconfig`?